extensions/http/src/main/java/jolie/net/HttpProtocol.java

```diff
@@ -681,7 +684,7 @@ && hasOperationSpecificParameter( message.operationName(), Parameters.STATUS_COD
 					" is sending a message with status code " +
 					statusCode +
 					", which is not in the HTTP specifications." );
-				statusCode = 500;
+				statusCode = DEFAULT_STATUS_CODE;
```

Comment:
Isn't this an internal server error? (The server is doing something that doesn't make sense because it is misconfigured.)

Reply:
Maybe we could add an exception to this method, throw it containing the new fault, and then catch it at an outer caller to change the comm message?